### PR TITLE
Add more NFRs to DevGuide

### PR DESCRIPTION
**Pull request by rish-16**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -494,11 +494,13 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-*{More to be added}*
+1. Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
+2. Should be able to hold up to 1000 patients without noticeable sluggishness in performance for typical usage.
+3. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+4. Should not require an external installer or launcher.
+5. Data should be stored locally in an easily-editable text file.
+6. Should not depend on any cloud-based remote server.
+7. Data should not be stored in an external Database Management System (DBMS) or data warehouse.
```
Comment (huyuxin0429) [inline docs/DeveloperGuide.md:503]: Double check if the numbering requires double spacing between the full stop and the beginning of the sentence? Seems like a common pattern in lists elsewhere in the document.

**Review by huyuxin0429 — COMMENTED** [review]
Non standard indentation for numbering lists, might be better to use double spacing between full stop and start of sentence.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (rish-16) [reply docs/DeveloperGuide.md]: Fixed in latest commit. Thanks.

**Review by joshenx — APPROVED** [review]
lgtm

**Pull request merged**